packages/cli/src/error-reporting.ts

```diff
@@ -1,106 +1,117 @@
+			environment,
+			enableTracing: false,
+			serverName,
+			beforeBreadcrumb: () => null,
+			beforeSend: (event, hint) => this.beforeSend(event, hint),
+			integrations: (integrations) => [
+				...integrations.filter(({ name }) => enabledIntegrations.includes(name)),
+				new RewriteFrames({ root: process.cwd() }),
+				new Integrations.RequestData({
+					include: {
+						cookies: false,
+						data: false,
+						headers: false,
+						query_string: false,
+						url: true,
+						user: false,
+					},
+				}),
+			],
+		});
+
+		setTag('server_type', this.instanceSettings.instanceType);
+
+		ErrorReporterProxy.init({
+			report: (error, options) => captureException(error, options),
+		});
+
+		this.initialized = true;
 	}
 
-	// Collect longer stacktraces
-	Error.stackTraceLimit = 50;
-
-	const {
-		N8N_VERSION: release,
-		ENVIRONMENT: environment,
-		DEPLOYMENT_NAME: serverName,
-	} = process.env;
-
-	const { init, captureException, setTag } = await import('@sentry/node');
-
-	const { RewriteFrames } = await import('@sentry/integrations');
-	const { Integrations } = await import('@sentry/node');
-
-	const enabledIntegrations = [
-		'InboundFilters',
-		'FunctionToString',
-		'LinkedErrors',
-		'OnUnhandledRejection',
-		'ContextLines',
-	];
-	const seenErrors = new Set<string>();
-
-	init({
-		dsn,
-		release,
-		environment,
-		enableTracing: false,
-		serverName,
-		beforeBreadcrumb: () => null,
-		integrations: (integrations) => [
-			...integrations.filter(({ name }) => enabledIntegrations.includes(name)),
-			new RewriteFrames({ root: process.cwd() }),
-			new Integrations.RequestData({
-				include: {
-					cookies: false,
-					data: false,
-					headers: false,
-					query_string: false,
-					url: true,
-					user: false,
-				},
-			}),
-		],
-		beforeSend(event, { originalException }) {
-			if (!originalException) return null;
-
-			if (originalException instanceof AxiosError) return null;
-
-			if (
-				originalException instanceof QueryFailedError &&
-				['SQLITE_FULL', 'SQLITE_IOERR'].some((errMsg) => originalException.message.includes(errMsg))
-			) {
-				return null;
-			}
-
-			if (originalException instanceof ApplicationError) {
-				const { level, extra, tags } = originalException;
-				if (level === 'warning') return null;
-				event.level = level;
-				if (extra) event.extra = { ...event.extra, ...extra };
-				if (tags) event.tags = { ...event.tags, ...tags };
-			}
-
-			if (originalException instanceof Error && originalException.stack) {
-				const eventHash = createHash('sha1').update(originalException.stack).digest('base64');
-				if (seenErrors.has(eventHash)) return null;
-				seenErrors.add(eventHash);
-			}
-
-			return event;
-		},
-	});
-
-	setTag('server_type', Container.get(InstanceSettings).instanceType);
-
-	ErrorReporterProxy.init({
-		report: (error, options) => captureException(error, options),
-	});
-
-	initialized = true;
-};
+	beforeSend(event: ErrorEvent, { originalException }: EventHint) {
```

Comment:
```suggestion
	private beforeSend(event: ErrorEvent, { originalException }: EventHint) {
```

Reply:
we have tests for this. can't make it private without creating a typing error.

Reply:
Never seen a good solution for this. Often it's useful to test private methods but that means TS exceptions :/